pkgs/build-support/fetchgit/default.nix

```diff
@@ -77,7 +78,7 @@ stdenvNoCC.mkDerivation {
   else
     lib.fakeSha256;
 
```

Comment:
Something that could be good to add is an assertion that sparse mode is enabled when cone mode is enabled

Reply:
So I'm thinking this right below the `deepClone -> leaveDotGit` assert:
```nix
assert nonConeMode -> (sparseCheckout != "");
```

Do you think that's good, or should I add a message like the `md5` assert?

Reply:
I think assert nonConeMode -> sparseCheckout should be good enough. Since it's a bool (or should be), the != "" doesn't seem necessary to me.

Reply:
`sparseCheckout` is a string, and it should be a string since that's how you specify the paths.

Reply:
oops, I thought about that the wrong way around :) you're right